dv_utils/solid_utils.py

```diff
@@ -0,0 +1,260 @@
+import requests
+import json
+import base64
+from os import environ
+from dv_utils import audit_log, LogLevel
+
+"""
+Fetches a resource from a pod using acp protocol and return as string
+"""
+def get_acp_from_pod_url(pod_url: str, path: str, solid_oidp_token: str = None, verify_grant: bool = False) -> str:
+  resource_uri = f"{pod_url}/{path}"
+
+  if not solid_oidp_token:
+    solid_oidp_token = __get_dv_soidp_token()
+
+  uma_token = get_uma_token(solid_oidp_token, resource_uri, verify_grant)
+
+  res = requests.get(resource_uri, headers={'Authorization': f"Bearer {uma_token}"})
+  if not res.ok:
+    audit_log(f"Could not get file with uma token. Got [{res.status_code}]: {res.text}")
+    return None
+  
+  return res.text
+
+def __get_dv_soidp_token() -> str:
+  token_endpoint = f"https://solid-idp.datavillage.me/token"  # TODO: how to configure solid idp?
+  webid = __get_cage_webid()
+  appid = __get_cage_appid()
+  user_name = environ['SOLID_IDP_USERNAME']
+  password = environ['SOLID_IDP_PASSWORD']
```

Comment:
Hmm hier gaan we nog iets op moeten vinden, waarschijnlijk wachten tot we die secretless auth hebben